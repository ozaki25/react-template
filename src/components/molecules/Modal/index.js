import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Dialog from 'src/components/atoms/dialogs/Dialog';

function Modal({
  title, children, buttonLabel, onClick,
}) {
  return (
    <Container>
      <DialogContainer>
        <Dialog title={title} buttonLabel={buttonLabel} onClick={onClick}>
          {children}
        </Dialog>
      </DialogContainer>
    </Container>
  );
}

const Container = styled.div`
  background-color: #000;
  border: none;
  height: 100%;
  left: 0;
  margin: 0;
  opacity: 0.6;
  position: fixed;
  top: 0;
  width: 100%;
`;

const DialogContainer = styled.div`
  margin-top: 70%;
  margin-left: 15px;
  margin-right: 15px;
  opacity: 1;
`;

Modal.propTypes = {
  title: PropTypes.string,
  children: PropTypes.string.isRequired,
  buttonLabel: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

Modal.defaultProps = {
  title: '',
  onClick: null,
};

export default Modal;
