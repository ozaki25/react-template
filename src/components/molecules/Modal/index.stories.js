import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import Modal from '.';

const props = {
  title: '通信エラー',
  onClick: action('click'),
  buttonLabel: 'OK',
};

const stories = storiesOf('molecules/Moadl', module);

stories.add(
  '通常パターン',
  withInfo('Modal')(() => (
    <Modal {...props}>{text('text', '通信できませんでした。リトライします。')}</Modal>
  )),
);
