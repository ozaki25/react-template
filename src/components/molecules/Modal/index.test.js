import React from 'react';
import { snapshot } from 'test/helpers';
import Modal from '.';

const props = {
  title: 'タイトル',
  onClick: jest.fn(),
  buttonLabel: 'OK',
  children: 'コンテンツ',
};

snapshot('Modal', <Modal {...props} />);
