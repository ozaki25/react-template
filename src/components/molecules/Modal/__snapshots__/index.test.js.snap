// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Modal 1`] = `
.c2 {
  background-color: #f5f5f5;
  border-radius: 4px;
  padding: 10px;
  text-align: center;
}

.c3 {
  margin-bottom: 5px;
  margin-top: 5px;
}

.c4 {
  font-size: 18px;
  font-weight: bold;
}

.c5 {
  margin-bottom: 5px;
  margin-top: 5px;
}

.c6 {
  font-size: 16px;
}

.c7 {
  margin-bottom: 5px;
  margin-top: 5px;
}

.c8 {
  background-color: #333;
  border-radius: 4px;
  color: #f5f5f5;
  font-size: 16px;
  height: 44px;
  width: 150px;
}

.c0 {
  background-color: #0009;
  border: none;
  height: 100%;
  left: 0;
  margin: 0;
  position: fixed;
  top: 0;
  width: 100%;
}

.c1 {
  margin-top: 70%;
  margin-left: 15px;
  margin-right: 15px;
}

<Modal
  buttonLabel="OK"
  onClick={[Function]}
  title="タイトル"
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <SmallDialog
            buttonLabel="OK"
            onClick={[Function]}
            title="タイトル"
          >
            <styled.div>
              <div
                className="c2"
              >
                <styled.div>
                  <div
                    className="c3"
                  >
                    <styled.p>
                      <p
                        className="c4"
                      >
                        タイトル
                      </p>
                    </styled.p>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="c5"
                  >
                    <styled.p>
                      <p
                        className="c6"
                      >
                        コンテンツ
                      </p>
                    </styled.p>
                  </div>
                </styled.div>
                <styled.div>
                  <div
                    className="c7"
                  >
                    <styled.button
                      onClick={[Function]}
                    >
                      <button
                        className="c8"
                        onClick={[Function]}
                      >
                        OK
                      </button>
                    </styled.button>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </SmallDialog>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Modal>
`;
